anypoint-signin-aware.js

```diff
@@ -680,6 +706,15 @@ export class AnypointSigninAware extends LitElement {
     AnypointAuth.clientId = newId;
   }
 
+  /**
+   * Sets AnypointAuth with an array of scopes, e.g. ['full','profile','email']
+   * @param {String} newScopes space separated scopes, e.g. 'full profile email'
+   */
+  _scopesChanged(newScopes) {
+    const scopes = newScopes && newScopes.split(' ');
+    AnypointAuth.scopes = scopes;
```

Comment:
And here's a problem with previously set scope and authorization token. 
Should the token be discarded when scope changes?
Situation 1:
Initial scope `profile email`
user: authorized
change scope to: `profile`

Situation 2:
Initial scope `profile`
user: authorized
change scope to: `profile email`

In scenario 1 there's no need to change the button or authentication state as new scope is still valid for the previous one.  No need to do anything. However, in scenario 2 new scope has been added and the user is no longer authorized for this scope. User action is required and button should change it's state to not signed in.
The second situation should inform the application that additional authentication is required (it is not the same as just authentication is required) so the application can perform background auth and/or inform the user about the change.

Reply:
I think it's better to inform the user of the change (user always needs to be informed of permission changes). Where would be the best place to set the signed in state to false when the token requests more state than was initially granted?